pkg/core/writer/signalfx/spans.go

```diff
@@ -21,9 +21,21 @@ func (sw *Writer) sendSpans(ctx context.Context, spans []*trace.Span) error {
 		// This sends synchonously
 		err := sw.client.AddSpans(context.Background(), spans)
 		if err != nil {
-			log.WithFields(log.Fields{
-				"error": err,
-			}).Error("Error shipping spans to SignalFx")
+			var meta log.Fields
+			if sw.conf.LogTraceSpansFailedToShip {
+				jsonEncodedSpans, _ := json.Marshal(spans)
+				meta = log.Fields{
+					"error":   err,
+					"payload": jsonEncodedSpans,
```

Comment:
I assume this case will be enabled for debugging purposes only so it is OK to log entire batches?

Reply:
yes, this option is only to be enabled when investigating `tooManySpansInTrace` as the backend gets rid of these